collector-http.go

```diff
@@ -214,10 +214,15 @@ func (c *HTTPCollector) send() error {
 	if c.reqCallback != nil {
 		c.reqCallback(req)
 	}
-	if _, err = c.client.Do(req); err != nil {
+	resp, err := c.client.Do(req)
+	if err != nil {
 		c.logger.Log("err", err.Error())
 		return err
 	}
+	resp.Body.Close()
+	if resp.StatusCode != http.StatusAccepted {
```

Comment:
This is too brittle. we could have different http collector server implementations which return 200 / 204, etc.

I'd only trigger the logger on a non 2xx code.